### PR TITLE
IG-14339

### PR DESCRIPTION
**Pull request by talIguaz**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,6 +17,8 @@ require (
 	github.com/v3io/frames v0.6.8-v0.9.11
 	github.com/v3io/v3io-go v0.0.5-0.20191205125653-9003ae83f0b6
 	github.com/xwb1989/sqlparser v0.0.0-20180606152119-120387863bf2
+	google.golang.org/genproto v0.0.0-20181026194446-8b5d7a19e2d9 // indirect
+	gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127 // indirect
```
Comment (gtopper) [inline go.mod]: Run `go mod tidy`?

File: pkg/tsdb/schema/schema.go [hunk 2]
```diff
@@ -161,7 +162,7 @@ func rateToHours(samplesIngestionRate string) (int, error) {
 		return 0, errors.Wrap(err, parsingError.Error())
 	}
 	if i <= 0 {
-		return 0, fmt.Errorf("Invalid samples ingestion rate (%s). The rate cannot have a negative number of samples.", samplesIngestionRate)
+		return 0, fmt.Errorf("invalid samples ingestion rate (%s). The rate cannot have a negative number of samples", samplesIngestionRate)
```
Comment (gtopper) [inline pkg/tsdb/schema/schema.go]: We already capitalized like all of them... It's too late now!

File: pkg/tsdb/schema/schema.go [hunk 3]
```diff
@@ -175,15 +176,22 @@ func rateToHours(samplesIngestionRate string) (int, error) {
...
 		return err
 	}
 
 	if dayMillis%duration != 0 && duration%dayMillis != 0 {
-		return errors.New("The aggregation granularity should be a divisor or a dividend of 1 day. Examples: \"10m\"; \"30m\"; \"2h\".")
+		return errors.New("the aggregation granularity should be a divisor or a dividend of 1 day. Examples: \"10m\"; \"30m\"; \"2h\".")
+	}
+
+	if hasAggregates {
+		partitionIntervalDuration, _ := utils.Str2duration(partitionInterval) // safe to ignore error since we create 'partitionInterval'
```
Comment (gtopper) [inline pkg/tsdb/schema/schema.go]: If it's so safe, I dare you to panic on it 😛

File: pkg/tsdb/schema/schema.go [hunk 4]
```diff
@@ -175,15 +176,22 @@ func rateToHours(samplesIngestionRate string) (int, error) {
...
+		if partitionIntervalDuration/duration > MaxV3ioArraySize {
+			return errors.New("the aggregation granularity is to close to the expected rate provided. Try increasing the granularity to get an aggregation performance impact")
```
Comment (gtopper) [inline pkg/tsdb/schema/schema.go]: to -> too

File: pkg/tsdb/schema/schema.go [hunk 5]
```diff
@@ -50,6 +47,10 @@ func newSchema(samplesIngestionRate, aggregationGranularity, aggregatesList stri
 		return nil, errors.Wrapf(err, "Failed to parse aggregates list '%s'.", aggregatesList)
 	}
 
+	if err := validateAggregatesGranularity(aggregationGranularity, partitionInterval, len(aggregates) > 0); err != nil {
+		return nil, errors.Wrapf(err, "Failed to parse aggregation granularity '%s'.", aggregationGranularity)
```
Comment (gtopper) [inline pkg/tsdb/schema/schema.go]: Parsing is not the only reason that `validateAggregatesGranularity` might return an error.

File: pkg/tsdb/schema/schema.go [hunk 6]
```diff
@@ -175,15 +176,22 @@ func rateToHours(samplesIngestionRate string) (int, error) {
...
+func validateAggregatesGranularity(aggregationGranularity string, partitionInterval string, hasAggregates bool) error {
 	dayMillis := 24 * int64(time.Hour/time.Millisecond)
 	duration, err := utils.Str2duration(aggregationGranularity)
 	if err != nil {
 		return err
 	}
 
 	if dayMillis%duration != 0 && duration%dayMillis != 0 {
-		return errors.New("The aggregation granularity should be a divisor or a dividend of 1 day. Examples: \"10m\"; \"30m\"; \"2h\".")
+		return errors.New("the aggregation granularity should be a divisor or a dividend of 1 day. Examples: \"10m\"; \"30m\"; \"2h\".")
```
Comment (gtopper) [inline pkg/tsdb/schema/schema.go]: We already capitalized like all of them... It's too late now!

File: pkg/tsdb/schema/schema.go (hunk 3 above)
Comment (talIguaz) [reply pkg/tsdb/schema/schema.go]: i tried real hard, couldn't...

Comment (gtopper) [reply pkg/tsdb/schema/schema.go]: Try harder!

**Pull request merged**